api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1694,18 +1694,72 @@ type AWSServiceEndpoint struct {
 }
 
 type AzurePlatformSpec struct {
+	// Credentials is the object containing the Azure credentials needed when creating and managing Azure cloud
+	// infrastructure resources
+	//
+	// +kubebuilder:validation:Required
 	Credentials corev1.LocalObjectReference `json:"credentials"`
-	// The cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
-	// +kubebuilder:validation:Enum=AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud
+
+	// Cloud is the cloud environment identifier, valid values could be found here: https://github.com/Azure/go-autorest/blob/4c0e21ca2bbb3251fe7853e6f9df6397f53dd419/autorest/azure/environments.go#L33
+	//
+	// +kubebuilder:validation:Enum="";AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud;AzureStackCloud
 	// +kubebuilder:default="AzurePublicCloud"
-	Cloud             string `json:"cloud,omitempty"`
-	Location          string `json:"location"`
+	Cloud string `json:"cloud,omitempty"`
+
+	// Location is Azure region in which the Azure cloud infrastructure resources will be created
+	//
+	// +immutable
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Location is immutable"
+	Location string `json:"location"`
+
+	// ResourceGroupName is the name of the resource group where any cloud infrastructure resources
+	// created by the Hosted Cluster are to be placed
+	//
+	// +immutable
+	// +kubebuilder:default:=default
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
 	ResourceGroupName string `json:"resourceGroup"`
-	VnetName          string `json:"vnetName"`
-	VnetID            string `json:"vnetID"`
-	SubnetName        string `json:"subnetName"`
-	SubscriptionID    string `json:"subscriptionID"`
+
+	// VnetName is the name of the VNET to place created subnets and VMs into
+	//
+	// +immutable
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="VnetName is immutable"
+	VnetName string `json:"vnetName"`
+
+	// VnetID is the ID of the VNET to place created subnets and VMs into
+	//
+	// +optional
```

Comment:
nit: `+optional` and `+immutable` usually go at the end right above the var name.